### PR TITLE
fix(spec): true up TC39 stages, spec anchors, and shipped-surface claims

### PR DESCRIPTION
**Pull request by frostney**
## Summary

- Release-prep truth-sync: every hand-maintained stage, spec anchor, and shipped-surface claim now matches the normative sources (tc39/proposals @ `600a4278`, ECMA-262 es2026, measured runner output, and the engine source).
- **Observable API change**: `Goccia.proposal` gains `stage-4` (Temporal) and `stage-2.7` (Decorators, Decorator Metadata), moves Enums to `stage-1` with the canonical tc39 URL, and **drops the now-empty `stage-0` key**. `Goccia.spec["2027"]` becomes Joint Iteration (merged into the ECMA-262 working draft); Temporal's dead `#sec-temporal-objects` anchor is replaced by the proposal spec.
- Docs: corrected five ECMA-262 section numbers and two dead anchors in `language.md`, four stage cells in `language-tables.md`, Temporal reframed from "ES2027" to Stage 4; `built-ins.md` documents the fractional stage keys.
- README/website: test count restated from measurement (12,313 tests / 1,551 files), five missing shipped globals added to the inventory (`queueMicrotask`, `structuredClone`, `atob`, `btoa`, `EventTarget`/`Event`), ShadowRealm row added to both profile-table mirrors, hand-typed 8 KiB caps now interpolate `MAX_GOCCIA_CODE_BYTES`.
- CONTEXT.md: eleven missing glossary terms added per domain-modeling (recommended profile, compatibility dashboard, the four CLI runner hosts, differential suite, semantics oracle, drop-in replacement, profile report, wrapper infrastructure failure); "import baseline" reconciled to "seed baseline" across live prose (ADRs untouched).
- YAML snapshot in `interpreter.md` restamped from a fresh rerun: 336/402 (83.6%) unchanged, now dated with engine SHA.
- Non-goals: no conformance-number policy changes (sentinel already enforces; 0 findings), no pin bumps, no generated-data changes.

Independently reviewed by a fresh-context agent against the TC39 MCP; all four review findings applied.

## Testing

- [x] Verified no regressions and confirmed the change in end-to-end JavaScript/TypeScript tests — `tests` and `tests --mode=bytecode` both 12,313/12,313 on the final diff; `tests/built-ins/Goccia` assertions hold against the new stage keys
- [x] Updated documentation — this PR is the documentation truth-sync; all five doc checks green (`check-doc-links`, `-symbols`, `-duplication`, `-length`, `check-conformance-claims`)
- [x] Optional: native Pascal surface — `format.pas --check` green; `Goccia.Spec.pas` restructure covered by `tests/built-ins/Goccia/proposal.js` + `spec.js`; website `bun test`/Biome/`tsc` green
- [ ] Optional: benchmarks — not applicable (no execution-path change)

## Linked issues

Related: #1071 (baseline evidence posted), #1116 (sentinel coverage audit posted)

### CODE REVIEW

File: CONTEXT.md [hunk 1]
```diff
@@ -76,7 +80,7 @@ _Avoid_: Capability audit event, rate limit, allowlist.
 
 **Seed baseline**:
 An explicitly imported snapshot used to initialise a sandbox-visible filesystem. Top-level sandbox seeds copy from host paths or inline seed config entries; nested child sandbox seeds copy from the parent virtual filesystem or inline child entries. A seed baseline is not a live mount and does not make the source path ambiently available to running source.
-_Avoid_: Mount, host filesystem access.
+_Avoid_: Mount, host filesystem access, import baseline.
```
[the remark this answers is not in the document]
Comment (frostney) [reply CONTEXT.md:83]: Dismissed: ADRs are immutable records per CONTRIBUTING.md ("Do not retroactively update ADRs"). The glossary's _Avoid_ entry is the historical-synonym documentation.

> [!NOTE]
> Created on behalf of @frostney using Fable 5.

File: README.md [hunk 2]
```diff
@@ -235,7 +236,7 @@ For custom providers, pass a JavaScript module to `--host-environment` or implem
 
 ### Run Tests
 
-GocciaScript has 11,000+ JavaScript unit tests covering language features, built-in objects, and edge cases.
+GocciaScript has 12,000+ JavaScript unit tests across 1,500+ test files, covering language features, built-in objects, and edge cases.
```
[the remark this answers is not in the document]
Comment (frostney) [reply README.md]: Fixed in c5b08994: the sentence now reads "JavaScript end-to-end tests", matching docs/testing.md's layer name.

> [!NOTE]
> Created on behalf of @frostney using Fable 5.

File: source/units/Goccia.Spec.pas [hunk 3]
```diff
@@ -178,7 +178,7 @@ implementation
   // ---------------------------------------------------------------------------
   ES2026_FEATURES: array[0..9] of TGocciaFeatureEntry = (
     (Name: 'JSON.parse Source Text Access';         Link: 'https://tc39.es/ecma262/#sec-json.parse'),
-    (Name: 'Explicit Resource Management';          Link: 'https://tc39.es/ecma262/#sec-using-declaration'),
+    (Name: 'Explicit Resource Management';          Link: 'https://tc39.es/ecma262/#sec-let-and-const-declarations'),
```
[the remark this answers is not in the document]
Comment (frostney) [reply source/units/Goccia.Spec.pas:181]: Dismissed: ERM shipped in ES2026 (your quoted result agrees: "not part of the ECMAScript 2027 draft"). In es2026 the using-declarations clause is §14.3.1 sec-let-and-const-declarations; sec-using-declaration does not exist (verified via TC39 MCP).

> [!NOTE]
> Created on behalf of @frostney using Fable 5.
